crypto/fipsmodule/rand/new_rand.c

```diff
@@ -50,10 +54,27 @@ static int rand_ensure_valid_state(void) {
   return 1;
 }
 
-// TODO
-// For UBE.
-static int rand_ensure_ctr_drbg_uniquness(struct rand_thread_local_state *state,
-  size_t out_len) {
+// rand_ensure_ctr_drbg_uniquness computes whether |state| must be randomized to
+// ensure uniqueness.
+//
+// Note: If |rand_ensure_ctr_drbg_uniquness| returns 0 it does not necessarily
+// imply that an UBE occurred. It can also mean that no UBE detection is
+// supported or that UBE detection failed. In these cases, |state| must also be
+// randomized to ensure uniqueness. Any special future cases can be handled in
+// this function. 
+//
+// Return 1 if |state| must be randomized. 0 otherwise.
+static int rand_ensure_ctr_drbg_uniquness(struct rand_thread_local_state *state) {
+
+  uint64_t current_generation_number = 0;
+  if (CRYPTO_get_ube_generation_number(&current_generation_number) != 1) {
+    return 0;
+  }
+
+  if (current_generation_number != state->generation_number) {
+    state->generation_number = current_generation_number;
+    return 0;
+  }
```

Comment:
Shouldn't this return 1? Isn't this saying the generation number is different because of an UBE? 
> Return 1 if |state| must be randomized. 0 otherwise.

Reply:
Yes, indeed, I screwed this up and you reviewed before I could add some tests (: